From 2b6f5b2a70158503c844e8259717b7168ced9bc0 Mon Sep 17 00:00:00 2001
From: Eric Kilmer <eric.d.kilmer@gmail.com>
Date: Sat, 30 Mar 2024 18:26:27 -0400
Subject: [PATCH 4/5] Fix opening sla files on Windows

Compressed files need to be opened as binaries when reading.
---
 Ghidra/Features/Decompiler/src/decompile/cpp/sleigh.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Ghidra/Features/Decompiler/src/decompile/cpp/sleigh.cc b/Ghidra/Features/Decompiler/src/decompile/cpp/sleigh.cc
index 471b99199..ba1a3b22c 100644
--- a/Ghidra/Features/Decompiler/src/decompile/cpp/sleigh.cc
+++ b/Ghidra/Features/Decompiler/src/decompile/cpp/sleigh.cc
@@ -560,7 +560,7 @@ void Sleigh::initialize(DocumentStorage &store)
     if (el == (const Element *)0)
       throw LowlevelError("Could not find sleigh tag");
     sla::FormatDecode decoder(this);
-    ifstream s(el->getContent());
+    ifstream s(el->getContent(), std::ios_base::binary);
     if (!s)
       throw LowlevelError("Could not open .sla file: " + el->getContent());
     decoder.ingestStream(s);
-- 
2.44.0

